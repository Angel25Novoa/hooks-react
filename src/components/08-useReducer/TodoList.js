import React from 'react'
import PropTypes from 'prop-types';
import TodoListItem from './TodoListItem'
import './styles.css'

const TodoList = ({todos, handleDelete, handleToggle}) => {
  return (
    <ul>
      {
        todos.map((todo, index) => {
          return (
            <TodoListItem 
              key={todo.id}
              todo={todo} 
              index={index} 
              handleDelete={handleDelete} 
              handleToggle={handleToggle}
            />
          )
        })
      }
    </ul>
  )
}

TodoList.propTypes = {
  todos: PropTypes.array.isRequired,
  handleDelete: PropTypes.func,
  handleToggle: PropTypes.func
}

export default TodoList