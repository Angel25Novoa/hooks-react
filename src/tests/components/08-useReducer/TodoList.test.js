import React from 'react'
import {shallow} from 'enzyme'
import TodoList from '../../../components/08-useReducer/TodoList'
import {demoState} from '../../fixtures/demoTodos'

describe((`TodoList component tests`), () => {
  const handleDelete = jest.fn() 
  const handleToggle = jest.fn()
  const wrapper = shallow(<TodoList 
    //AsÃ­ pasamos los argumentos
    todos = {demoState}
    handleDelete = {handleDelete}
    handleToggle = {handleToggle}
  />)
  test((`TodoList component should render without errors`), () => {
    expect(wrapper).toMatchSnapshot()
  })
  test((`TodoList component should render a list of TodoListItem`), () => {
    expect(wrapper.find(`TodoListItem`).length).toBe(demoState.length)
    expect(wrapper.find(`TodoListItem`).at(0).prop(`handleDelete`)).toEqual(expect.any(Function))
  })
})