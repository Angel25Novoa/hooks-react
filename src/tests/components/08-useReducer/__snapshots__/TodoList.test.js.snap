// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TodoList component tests TodoList component should render without errors 1`] = `
<ul>
  <TodoListItem
    handleDelete={[MockFunction]}
    handleToggle={[MockFunction]}
    index={0}
    key="1"
    todo={
      Object {
        "done": false,
        "id": 1,
        "name": "Hacer la compra",
      }
    }
  />
  <TodoListItem
    handleDelete={[MockFunction]}
    handleToggle={[MockFunction]}
    index={1}
    key="2"
    todo={
      Object {
        "done": false,
        "id": 2,
        "name": "Hacer el trabajo",
      }
    }
  />
</ul>
`;
